### PR TITLE
CLN/TST: test_series_align_multiindex_containing_nan

### LINKED ISSUE #49443: TST: Refactor ``test_subtracting_two_series_with_unordered_index_and_all_nan_index``
The test is hard to read and hard to understand. We should refactor it to make it more straightforward without those parametrizations. Pending #49442

### PR DESCRIPTION
**Pull request by lukemanley**
- [x] closes #49443
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.

### CODE REVIEW

File: pandas/tests/indexing/multiindex/test_multiindex.py [hunk 1]
```diff
@@ -151,57 +151,33 @@ def test_rename_multiindex_with_duplicates(self):
...
+        result1, result2 = ser1.align(ser2)
+
+        mi = MultiIndex.from_arrays([[81.0, 82.0, np.nan], [np.nan, np.nan, np.nan]])
+        expected1 = Series([1.0, np.nan, 2.0], index=mi)
+        expected2 = Series([np.nan, 2.0, 1.0], index=mi)
+
+        tm.assert_series_equal(result1, expected1)
+        tm.assert_series_equal(result2, expected2)
+
+        mi1 = MultiIndex.from_arrays([[81.0, np.nan], [np.nan, np.nan]])
```
Comment (mroeschke) [inline pandas/tests/indexing/multiindex/test_multiindex.py:169]: From the issue, I think this block should be a separate test

Comment (lukemanley) [reply pandas/tests/indexing/multiindex/test_multiindex.py:171]: Ok, updated to split into two tests.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lukemanley

**Comment by phofl** [comment]
Thx as well, this was really hard to understand before

**Pull request merged**